proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+
+    public SDLTextAlignment getTextAlignment() {}
+
+    public MetadataType getTextField1Type() {}
+
+    public MetadataType getTextField2Type() {}
+
+    public MetadataType getTextField3Type() {}
+
+    public MetadataType getTextField4Type() {}
+
+    public ArrayList<SoftButtonObject> getSoftButtonObjects() {}
+
+    // Updates
+
+    public void beginUpdates(){}
+
+    public void endUpdates(CompletionListener listener){}
+
+}
+
+```
+
+##### SoftButtonManager
+
+This is a non-developer facing submanager of the screen manager. It maintains the softbuttons for the current template including their states and images needed.
+
+```java
+public class SoftButtonManager extends BaseSubManager {
+
+    protected boolean batchUpdates;
+
+    // Constructors
+
+    public SoftButtonManager(ISdl internalInterface, FileManager fileManager) {}
+
+    // Update Methods
+
+    public void update() {}
+
+    // Setters / Getters
+
+    public void setSoftButtonObjects(ArrayList<SoftButtonObject> softButtonObjects){}
+
+    public ArrayList<SoftButtonObject> getSoftButtonObjects() {}
+    
+    public void setHmiLevel(HMILevel level){}
+
+    public HMILevel getHmiLevel() {}
+
+    public void setCurrentMainField1(String mainField1){}
+
+    public String getCurrentMainField1(){}
+}
+```
+
+The states of softbuttons allow the developer to not have to manage multiple softbuttons that have very similar functionality. For example, a repeat button in a music app can be thought of as one button with three typical states: repeat off, repeat 1, and repeat on. So the developer really only wants to manage simple states between a single button. We introduce a class that handles these:
+
+```java
+public class SoftButtonObject {
+
+    public SoftButtonObject(String name, ArrayList<SoftButtonState> states, String initialStateName) {}
+
+    public SoftButtonObject(String name, SoftButtonState state, String initialStateName) {}
+
+    public Boolean transitionToStateNamed(String stateName){}
+
+    public void transitionToNextState() {}
+
+}
+```
+
+The `SoftButtonState` class is a small class that defines an individual state.
+
+```java
+class SoftButtonState {
+
+    SoftButtonState(String name, String text, SDLFile artwork) {}
+
+}
+```
+
+##### TextAndGraphicManager
+
+This submanager is based off and mimics the previously [accepted proposal and implementation for iOS](https://github.com/smartdevicelink/sdl_ios/blob/master/SmartDeviceLink/SDLTextAndGraphicManager.m). This is very similar to the soft button manager as it is a submanager of the screen manager that intends to be non-developer facing. Its purpose is to handle the logic of setting text and artworks. The text setting is dynamic, just as it is in iOS. The complexity of this seemingly simple problem does warrant a specific manager to help scope the code and responsibilities. The merits for this decision were also laid out in the aforementioned proposal and its review. 
```

Comment:
Can you include a link to the proposal you're referencing, in addition to the implementation link?